STAN.CLIENT/StanConnection.cs

```diff
@@ -86,14 +88,14 @@ internal void wait(int timeout)
                 {
                     Monitor.Wait(cond, timeout);
                 }
+                if (ex != null)
+                    throw ex;

```

Comment:
Related but not, we should likely switch to [`ExceptionDispatchInfo`](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.exceptionservices.exceptiondispatchinfo?view=netframework-4.8) for these so that we have a better understanding of the stack trace when the exception occurred.

Reply:
I agree, but would best be in a separate PR.